### ISSUE TITLE
Include mapaddress text

### ISSUE THREAD

**Issue by BigFav** [juliomalegria/python-craigslist#75]
Geocodes appear to be approximates in a lot of cases. In certain instances, the listing will have the address in the `mapaddress` div tag and it doesn't seem like this reads that. This is a feature request to include that information.

Example page - https://york.craigslist.org/apa/d/york-room-for-rent/7104679926.html
Example return:

```
{
  'id': '7104679926',
  'repost_of': None,
  'name': 'Room for Rent',
  'url': 'https://york.craigslist.org/apa/d/york-room-for-rent/7104679926.html',
  'datetime': '2020-05-07 08:47',
  'last_updated': '2020-05-07 08:47',
  'price': '$669',
  'where': None,
  'has_image': True,
  'geotag': (39.955682, -76.718121)
}
```

Page with inspection open:
![Image with inspection](https://user-images.githubusercontent.com/2354042/83319871-f274f780-a1f6-11ea-857b-5c4c41f28e8b.png)

**Comment by irahorecka** [comment]
Unfortunately, I believe this is a craigslist.org <http://craigslist.org/> problem.

**Comment by BigFav** [comment]
This is just a scraper, which is why I identified the tag that displays this. This tag is in the output of the following code (which is found in this package):
``` python
import requests
from bs4 import BeautifulSoup

response = requests.get('http://york.craigslist.org/apa/d/york-room-for-rent/7104679926.html')
soup = BeautifulSoup(response.content, 'html.parser')
print(soup)
```

Unclear how this is a craigslist problem given it is output from the craigslist request call.

**Comment by irahorecka** [comment]
This is not this library's problem as it is Craigslist's. Check out the html source for the [link](https://york.craigslist.org/apa/d/york-room-for-rent/7104679926.html) you provided (btw the link will expire eventually).

[html source](https://i.imgur.com/y4uOvdd.png)

**Comment by BigFav** [comment]
As stated above, this issue is to include the `mapaddress` div tag in the result, which in this case is "239 E Boundary Ave". Please look at what is highlighted in the original image on the inspection frame for reference.

**Comment by juliomalegria** [comment]
Hey @BigFav,
I think this would be a great addition. Thanks for the feature request! Seems pretty simple to do, I'll work on it today or tomorrow.
However, just FYI, this won't get the lat,lng from that address. The geotag will remain the same, from the approx info provided by Craigslist, however there'll be a new field "address" is the `mapaddress` is present. Your example return was a bit confusing, but it would be something like this:

```
{
  'id': '7104679926',
  'repost_of': None,
  'name': 'Room for Rent',
  'url': 'https://york.craigslist.org/apa/d/york-room-for-rent/7104679926.html',
  'datetime': '2020-05-07 08:47',
  'last_updated': '2020-05-07 08:47',
  'price': '$669',
  'where': None,
  'has_image': True,
  'geotag': (39.955682, -76.718121),
  'address': '239 E Boundary Ave'.         <<<< NEW
}
```

Thanks @irahorecka for stepping in to help 👍

**Comment by BigFav** [comment]
Definitely doesn't map sense to be used as `geotag`, was just noting it is more accurate than `geotag`. Where does `where` come from? From my naive perspective, it seems like it could go there.

**Comment by juliomalegria** [comment]
I had forgotten about `where`. This comes from `result-hood` in the list results. In the US, this always seems to be only the neighborhood, for e.g.

![Screen Shot 2020-06-03 at 9 47 40 PM](https://user-images.githubusercontent.com/1008637/83682694-90e0ce80-a5e4-11ea-94be-cb293eb977dd.png)

However in other countries, `result-hood` seems to contain the full address:

![Screen Shot 2020-06-03 at 9 49 32 PM](https://user-images.githubusercontent.com/1008637/83682744-a229db00-a5e4-11ea-9843-646ab7270690.png)

And in others (for example here in Berlin) is a mix of both: only the city/neighborhood, or the full address:

![Screen Shot 2020-06-03 at 9 51 32 PM](https://user-images.githubusercontent.com/1008637/83682865-cab1d500-a5e4-11ea-8cb1-b4809e9feabf.png)

I'm hesitating whether to just override `where` with the value of `mapaddress` if it exists, or to have 2 separate fields. I guess option C would be to concatenate them. For example in: 

https://sfbay.craigslist.org/pen/apa/d/san-bruno-updated-1br-1ba-with-parking/7134532065.html

mapaddress is "443 San Anselmo Ave N near Mastick and Martin" and where is "san bruno". The resulting `where` could be: "443 San Anselmo Ave N near Mastick and Martin, San Bruno".

Any thoughts?

**Comment by BigFav** [comment]
Two separate fields is the simplest to me as it puts the power in the hands of the user; this tool is simply a scraper. My question was more for information purposes, apologies for the confusion.

**Comment by KeeonTabrizi** [comment]
@juliomalegria wondering if there's any progress on implementing this new extracted object. I'm considering doing something similar but if the work is already in flight think it makes sense to check in.

**Comment by juliomalegria** [comment]
Key @KeeonTabrizi, I'm sorry. This change is very much in flight. As a matter of fact, it was ready, I had just forgotten to push it. I'll do it now and release a new version.

**Comment by juliomalegria** [comment]
This was resolved in commit 1f8c44c. I've just released version `1.0.9` in PyPI
